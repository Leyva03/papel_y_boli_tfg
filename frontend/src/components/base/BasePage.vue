<template>
    <div class="p-6 max-w-2xl mx-auto">
        <h1 class="titulo-inicio text-4xl titulo-sombra-suave mb-4 text-center text-white">{{ title }}</h1>
      <slot></slot>
    </div>
  </template>
  
  <script setup>
  defineProps({
    title: String
  })
  </script>
  