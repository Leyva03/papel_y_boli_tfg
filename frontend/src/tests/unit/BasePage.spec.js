import { mount } from '@vue/test-utils';
import BasePage from '@/components/base/BasePage.vue';

describe('BasePage.vue', () => {
  it('debería renderizar el título proporcionado correctamente', () => {
    const titleText = 'Mi Título de Prueba';
    const wrapper = mount(BasePage, {
      props: {
        title: titleText,
      },
    });
    const h1 = wrapper.find('h1.titulo-inicio');
    expect(h1.exists()).toBe(true);
    expect(h1.text()).toBe(titleText);
  });

  it('debería renderizar el contenido del slot por defecto', () => {
    const slotContent = '<p>Este es el contenido del slot.</p>';
    const wrapper = mount(BasePage, {
      props: {
        title: 'Título con Slot',
      },
      slots: {
        default: slotContent,
      },
    });
    expect(wrapper.html()).toContain(slotContent);
    expect(wrapper.find('p').text()).toBe('Este es el contenido del slot.');
  });

  it('debería coincidir con el snapshot con título y contenido de slot', () => {
    const titleText = 'Título para Snapshot';
    const slotContent = '<div>Contenido del slot para snapshot</div>';
    const wrapper = mount(BasePage, {
      props: {
        title: titleText,
      },
      slots: {
        default: slotContent,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('debería coincidir con el snapshot solo con el título (sin slot)', () => {
    const titleText = 'Título Solo para Snapshot';
    const wrapper = mount(BasePage, {
      props: {
        title: titleText,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});