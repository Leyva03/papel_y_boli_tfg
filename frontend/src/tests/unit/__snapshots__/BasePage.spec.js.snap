// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePage.vue debería coincidir con el snapshot con título y contenido de slot 1`] = `
"<div class="p-6 max-w-2xl mx-auto">
  <h1 class="titulo-inicio text-4xl titulo-sombra-suave mb-4 text-center text-white">Título para Snapshot</h1>
  <div>Contenido del slot para snapshot</div>
</div>"
`;

exports[`BasePage.vue debería coincidir con el snapshot solo con el título (sin slot) 1`] = `
"<div class="p-6 max-w-2xl mx-auto">
  <h1 class="titulo-inicio text-4xl titulo-sombra-suave mb-4 text-center text-white">Título Solo para Snapshot</h1>
</div>"
`;
